lib/nats.js

```diff
@@ -1884,27 +1884,65 @@ Client.prototype.requestOne = function(subject, opt_msg, opt_options, timeout, c
     if (this.options.useOldRequestStyle) {
         return this.oldRequestOne(subject, opt_msg, opt_options, timeout, callback);
     }
+    // eslint-disable-next-line prefer-rest-params
+    const args = callbackShifter(5, Array.prototype.slice.call(arguments));
+    subject = args[0];
+    opt_msg = args[1];
+    opt_options = args[2];
+    timeout = args[3];
+    callback = args[4];
+
+    if (timeout === undefined && typeof opt_options === "number") {
+        timeout = opt_options;
+        opt_options = undefined;
+    }
 
-    if (typeof opt_msg === 'number') {
-        callback = opt_options;
+    if (timeout === undefined && opt_options === undefined && typeof opt_msg === "number") {
         timeout = opt_msg;
-        opt_options = null;
         opt_msg = EMPTY;
     }
 
-    if (typeof opt_options === 'number') {
-        callback = timeout;
-        timeout = opt_options;
-        opt_options = null;
-    }
-
     opt_options = opt_options || {};
     opt_options.max = 1;
     opt_options.timeout = timeout;
 
     return this.request(subject, opt_msg, opt_options, callback);
 };
 
+// finds the last index that satisfies the testFn in array a
+function lastIndexOf(a, testFun) {
+    const idx = a.length-1;
+    for(let i=idx; i > -1; i--) {
+        if (testFun(a[i])) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+// returns an array that shifts the last function in the
+// array to the last element of the array
+function callbackShifter(maxArgs, args) {
+    // last argument is expected to be a function
+    const a = new Array(maxArgs);
+    const idx = lastIndexOf(args, (v) => typeof v === 'function');
+
+    // found a function, at the last index
+    if (idx !== -1) {
+        a[maxArgs - 1] = args[idx];
+        for (let i = 0; i < idx; i++) {
+            a[i] = args[i];
+        }
+    } else {
+        for (let i=0; i < args.length; i++) {
```

Comment:
Did you not say that you could return args here?

Reply:
rethought that at least we get a sanitized version of the args - (it will be the expected length regardless of what they passed)